librz/analysis/p/analysis_tricore_cs.c

```diff
@@ -0,0 +1,799 @@
+	case TRICORE_INS_SYSCALL:
+	case TRICORE_INS_DISABLE:
+	case TRICORE_INS_ENABLE:
+	case TRICORE_INS_SVLCX:
+	case TRICORE_INS_WAIT:
+	case TRICORE_INS_RESTORE: {
+		op->type = RZ_ANALYSIS_OP_TYPE_SWI;
+		break;
+	}
+
+	case TRICORE_INS_CACHEA_I:
+	case TRICORE_INS_CACHEA_WI:
+	case TRICORE_INS_CACHEA_W:
+	case TRICORE_INS_CACHEI_I:
+	case TRICORE_INS_CACHEI_WI:
+	case TRICORE_INS_CACHEI_W:
+
+	case TRICORE_INS_CLO_B:
+	case TRICORE_INS_CLO_H:
+	case TRICORE_INS_CLO:
+	case TRICORE_INS_CLS_B:
+	case TRICORE_INS_CLS_H:
+	case TRICORE_INS_CLS:
+	case TRICORE_INS_CLZ_B:
+	case TRICORE_INS_CLZ_H:
+	case TRICORE_INS_CLZ: {
+		// Supervisor mode only
+		op->type = RZ_ANALYSIS_OP_TYPE_UNK;
+		break;
+	}
+
+	case TRICORE_INS_CRC32B_W:
+	case TRICORE_INS_CRC32L_W:
+	case TRICORE_INS_CRC32_B:
+	case TRICORE_INS_CRCN: {
+		op->type = RZ_ANALYSIS_OP_TYPE_CRYPTO;
+		op->family = RZ_ANALYSIS_OP_FAMILY_CRYPTO;
+		break;
+	}
+	case TRICORE_INS_CALLA:
+	case TRICORE_INS_CALLI:
+	case TRICORE_INS_CALL: {
+		op->type = RZ_ANALYSIS_OP_TYPE_CALL;
+		break;
+	}
+	case TRICORE_INS_DIV_U:
+	case TRICORE_INS_DIV: {
+		op->type = RZ_ANALYSIS_OP_TYPE_DIV;
+		break;
+	}
+	case TRICORE_INS_DEBUG:
+	case TRICORE_INS_NOP: {
+		op->type = RZ_ANALYSIS_OP_TYPE_NOP;
+		break;
+	}
+	case TRICORE_INS_NOR_T:
+	case TRICORE_INS_NOR: {
+		op->type = RZ_ANALYSIS_OP_TYPE_NOR;
+		break;
+	}
+
+	case TRICORE_INS_DEXTR:
+	case TRICORE_INS_EXTR_U:
+	case TRICORE_INS_EXTR:
+	case TRICORE_INS_INSERT:
+	case TRICORE_INS_INSN_T:
+	case TRICORE_INS_INS_T: {
+		op->type = RZ_ANALYSIS_OP_TYPE_REG;
+		break;
+	}
+	case TRICORE_INS_DIFSC_A: {
+		op->type = RZ_ANALYSIS_OP_TYPE_UNK;
+		break;
+	}
+
+	case TRICORE_INS_DSYNC: {
+		op->type = RZ_ANALYSIS_OP_TYPE_SYNC;
+		break;
+	}
+	case TRICORE_INS_DVADJ:
+	case TRICORE_INS_DVINIT_BU:
+	case TRICORE_INS_DVINIT_B:
+	case TRICORE_INS_DVINIT_HU:
+	case TRICORE_INS_DVINIT_H:
+	case TRICORE_INS_DVINIT_U:
+	case TRICORE_INS_DVINIT:
+	case TRICORE_INS_DVSTEP_U:
+	case TRICORE_INS_DVSTEP:
+
+	case TRICORE_INS_IMASK: {
+		op->type = RZ_ANALYSIS_OP_TYPE_UNK;
+		break;
+	}
+	case TRICORE_INS_ISYNC: {
+		op->type = RZ_ANALYSIS_OP_TYPE_SYNC;
+		break;
+	}
+
+	case TRICORE_INS_IXMAX_U:
+	case TRICORE_INS_IXMAX:
+	case TRICORE_INS_IXMIN_U:
+	case TRICORE_INS_IXMIN: {
+		op->type = RZ_ANALYSIS_OP_TYPE_UNK;
+		break;
+	}
+	case TRICORE_INS_JA:
+	case TRICORE_INS_JEQ_A:
+	case TRICORE_INS_JEQ:
+	case TRICORE_INS_JGEZ:
+	case TRICORE_INS_JGE_U:
+	case TRICORE_INS_JGE:
+	case TRICORE_INS_JGTZ:
+	case TRICORE_INS_JI:
+	case TRICORE_INS_JLA:
+	case TRICORE_INS_JLEZ:
+	case TRICORE_INS_JLI:
+	case TRICORE_INS_JLTZ:
+	case TRICORE_INS_JLT_U:
+	case TRICORE_INS_JLT:
+	case TRICORE_INS_JL:
+	case TRICORE_INS_JNED:
+	case TRICORE_INS_JNEI:
+	case TRICORE_INS_JNE_A:
+	case TRICORE_INS_JNE:
+	case TRICORE_INS_JNZ_A:
+	case TRICORE_INS_JNZ_T:
+	case TRICORE_INS_JNZ:
+	case TRICORE_INS_JZ_A:
+	case TRICORE_INS_JZ_T:
+	case TRICORE_INS_JZ:
+	case TRICORE_INS_J: {
```

Comment:
You will need to calculate jump offset, also some of them are conditional, should have a different type,